task/fbc-validation/0.1/fbc-validation.yaml

```
@@ -172,6 +177,25 @@ spec:
           failure_num=`expr $failure_num + 1`
           TESTPASSED=false
         fi
+        if [[ "${OCP_VER_MAJOR}" -ge "${OCP_BUNDLE_METADATA_THRESHOLD_MAJOR}" ]] && [[ "${OCP_VER_MINOR}" -ge "${OCP_BUNDLE_METADATA_THRESHOLD_MINOR}" ]]; then
+          #
+          # above the version threshold for mandatory `olm.csv.metadata` bundle metadata format, so presence of `olm.bundle.object` is an error
+          #
+          if ! ${OPM_BINARY} render ."${conffolder}" | jq -en 'reduce( inputs | select(.schema == "olm.bundle" and .properties[].type == "olm.bundle.object")) as $_ (0;.+1) == 0'; then
+            echo "!FAILURE! - olm.bundle.object is not permitted in a FBC fragment for OCP version ${OCP_VER_MAJOR}.${OCP_VER_MINOR}. Fragments must move to olm.csv.metadata bundle metadata."
```

```suggestion
            echo "!FAILURE! - olm.bundle.object bundle properties are not permitted in a FBC fragment for OCP version ${OCP_VER_MAJOR}.${OCP_VER_MINOR}. Fragments must move to olm.csv.metadata bundle metadata."
```

Should we also include something like this:
```
Use opm >= 1.46.0 with `--migrate-level=bundle-object-to-csv-metadata`
```

---

I didn't want to too-closely couple the tools, so I stopped short of providing guidance on how to resolve the issue.  I presumed that the first thing they would do is do a `opm $command --help` for the opm roles in production of their fragment.

---

But I think people will need to be told:
1. It is (almost always) `opm` that controls which of these properties is written.
2. _If_ you are using `opm`, we recommend 1.46.0 with `--migration-level` pinning.

---

I believe I resolved these comments with details in the new TROUBLESHOOTING.md file. @joelanford PTAL, and let me know if these are not resolved.